scripts/corr2_bengine_status.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# pylint: disable-msg=C0103
+# pylint: disable-msg=C0301
+"""
+View the status of a given Bengine.
+
+Created on Fri Jan  3 10:40:53 2014
+
+@author: paulp
+"""
+import sys
+import time
+import argparse
+import signal
+import os
+
+from casperfpga import utils as fpgautils
+from casperfpga import katcp_fpga
+from casperfpga import dcp_fpga
+import casperfpga.scroll as scroll
+from corr2 import utils
+
+parser = argparse.ArgumentParser(
+    description='Display information about a MeerKAT x-engine.',
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--hosts', dest='hosts', type=str,
+                    action='store', default='',
+                    help='comma-delimited list of hosts, or a corr2 '
+                         'config file')
+parser.add_argument('-p', '--polltime', dest='polltime',
+                    action='store', default=1, type=int,
+                    help='time at which to poll x-engine data, in seconds')
+parser.add_argument('--comms', dest='comms', action='store',
+                    default='katcp', type=str,
+                    help='katcp (default) or dcp?')
+parser.add_argument('--loglevel', dest='log_level',
+                    action='store', default='',
+                    help='log level to use, default None, options INFO, '
+                         'DEBUG, ERROR')
+args = parser.parse_args()
+
+def signal_handler(signal_, frame):
+    fpgautils.threaded_fpga_function(fpgas, 10, 'disconnect')
+    scroll.screen_teardown()
+    sys.exit(0)
+signal.signal(signal.SIGINT, signal_handler)
+signal.signal(signal.SIGHUP, signal_handler)
+
+polltime = args.polltime
+
+if args.log_level != '':
+    import logging
+    log_level = args.log_level.strip()
+    try:
+        logging.basicConfig(level=eval('logging.%s' % log_level))
```

Comment:
`eval()` is considered dangerous. You could rather use `getattr(logging, log_level)`